dev/io.openliberty.microprofile.telemetry.2.0.logging.internal/src/io/openliberty/microprofile/telemetry20/logging/internal/MpTelemetryLogMappingUtils.java

```diff
@@ -394,8 +404,33 @@ else if (value instanceof Integer)
         // Set the Attributes to the builder.
         builder.setAllAttributes(attributes.build());
 
-        // Set the Span and Trace IDs from the current context.
-        builder.setContext(Context.current());
+        // Set the Span and Trace IDs from the current context. We're not on the same thread at the point when access logs are collected
+        // so we need to extract the trace/span ID from the 'traceparent' request header.
+        if (customSpan != null) {
+            builder.setContext(Context.current().with(customSpan));
+        } else
+            builder.setContext(Context.current());
+
+    }
+
+    private static Span createSpan(String requestHeader, int propagator) {
+        SpanContext customSpanContext = null;
+        if (propagator == 0) { // Check the w3c format for the "traceparent" header
```

Comment:
Instead of checking it by values, wondering if it would be cleaner to just check it by the key name in the if/else. Easier to understand.